common/src/main/java/com/redhat/lightblue/rest/logging/LoggingFilter.java

```diff
@@ -0,0 +1,65 @@
+package com.redhat.lightblue.rest.logging;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+@WebFilter(urlPatterns = {"/*"})
+public class LoggingFilter implements Filter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingFilter.class);
+
+    public static final String HEADER_REQUEST_UUID = "RequestUUID";
+
+    @Override
+    public void destroy() {
+        //Do Nothing!
+    }
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain) throws IOException, ServletException {
+        try {
+            String requestUUID = UUID.randomUUID().toString();
```

Comment:
This returns a rather large hard to read garbage. Instead, why not have a static AtomicInteger with addAndGet()? That might cause problems with ,multiple hosts, so maybe a static hostname variable that is initialized once, concatenated with that int? So it becomes lightbluecrud1-1234, which might be more human readable.